README.md

```diff
@@ -0,0 +1,109 @@
+1. **arbitrary smartness**
+   * In order to fix issues with the type system and make numbers more easily usable by converting "strings" into integers/floats, what worked in one API broke another.
+1. **it's cumbersome to actually use a CLI**
+   * Even though authentication was dealt with nicely for the most part, actually using APIs required them to be enabled via the developer console. From there one would download a file and deposit it in the right spot. Only then one could start using the CLI.
+1. **oddly keyed login tokens stored on disk per scope**
+   * due to tokens being hashed by the scope(s) they represent, choosing a different scope forced you to re-authenticate, even though another token file already included the scope you wanted.
+1. **it took at least 6 weeks to get the first release on crates.io**
+   * development wasn't the fastest, and I claim one was slowed down due to too much manual testing.
+1. **there was no way to use the CI to test CLIs to actually interact with Google APIs**
+   * This was due to API usage being bound to a person, and these credentials were nothing you would want to have lying around in a public git repository.
+   * Not being able to test certain feature automatically and repeatedly takes time and reduces quality guarantees.
+1. **tests were minimal and most testing was like "if it compiles, it's good"**
+   * The _OP_ suffered from only having a few tests, and even though Rust only compiles decent quality software, by nature, certain reasoning was just in my head. _'Why is it doing this particular thing? It seems wrong'_ would be impossible to know. 
+   * manual testing is slow and error prone
+1. **Versions like 1.2.0+2019-08-23...` would additionally show the version of the Google API description, but is ignored by `cargo`** 
+   * This was done to discriminate the 'code' version from the version of the API it represents.
+   * As the `+` is ignored by `cargo`, to re-release a crate with a new version of the API, one would have to increment the patch level. However, that would force all crates to be re-released, even if their API version didn't change at all.
+   * This caused unnecessary spamming of `crates.io`, and the `+` should be a `-` to fix this.
+1. **The 'fields' parameter could not be used to have smaller response types**
+   * Some [Response Types](https://docs.rs/google-sheets4/1.0.10+20190625/google_sheets4/struct.Response.html) are huge, even though with the right `field` setting, one would only receive a fraction of the data. However, one would always have to allocate big structures to with most of the optional fields set to `None`.
+   * 💡Idea 💡: Can [serde(flatten)](https://serde.rs/field-attrs.html#flatten) be used to subdivide the possible field sets in the data structure? Probably it's not known which actual fields belong to each `field` argument.
```

Comment:
I'm not sure how serde(flatten) could help here because we're just dealing with a lack of information from the discovery service. However, if the generation library was written to allow per API customizations, I think this would be a great use of that. I'm not sure how far down that rabbit hole we would want to go as it may lead to a never ending series of changes, but it would be pretty easy to add a customization step in the creation of the API definition. Sheets could then use it to modify Response from an object to a anyOf (along with adding anyOf support which doesn't currently exist).